storagenode/contact/chore.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package contact
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+	"gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/internal/sync2"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/transport"
+	"storj.io/storj/satellite/overlay"
+	"storj.io/storj/storagenode/trust"
+)
+
+var mon = monkit.Package()
+
+// Config contains configurable parameters for contact chore
+type Config struct {
+	Interval time.Duration `help:"how frequently the node contact chore should run" releaseDefault:"1h" devDefault:"30s"`
+	// MaxSleep should remain at default value to decrease traffic congestion to satellite
+	MaxSleep time.Duration `help:"maximum duration to wait before pinging satellites" releaseDefault:"45m" devDefault:"0s" hidden:"true"`
+}
+
+// Chore is the contact chore for nodes announcing themselves to their trusted satellites
+type Chore struct {
+	log       *zap.Logger
+	self      overlay.NodeDossier
+	transport transport.Client
+
+	trust *trust.Pool
+
+	maxSleep time.Duration
+	Loop     *sync2.Cycle
+}
+
+// NewChore creates a new contact chore
+func NewChore(log *zap.Logger, interval time.Duration, maxSleep time.Duration, trust *trust.Pool, transport transport.Client, self overlay.NodeDossier) *Chore {
+	return &Chore{
+		log:       log,
+		self:      self,
+		transport: transport,
+
+		trust: trust,
+
+		maxSleep: maxSleep,
+		Loop:     sync2.NewCycle(interval),
+	}
+}
+
+// Run the contact chore on a regular interval with jitter
```

Comment:
not sure if we should expand on this comment a little bit and explain what is meant by jitter